### PR TITLE
Fix depwarn of read_rda

### PR DESCRIPTION
**Pull request by andreasnoack**

### CODE REVIEW

**Comment by andreasnoack** [comment]
We need a new version of `Compat.jl` before this will pass. https://github.com/JuliaLang/METADATA.jl/pull/6263

**Comment by ararslan** [comment]
This is awesome. Thanks so much, and sorry for the mess!

**Comment by ararslan** [comment]
> We need a new version of Compat.jl before this will pass.

Once that's tagged we should set 0.8.9 as the lower bound on Compat in REQUIRE.

**Comment by andreasnoack** [comment]
I'll do it as part of this PR.

**Comment by ararslan** [comment]
Right, that's what I meant

File: test/REQUIRE [hunk 1]
```diff
@@ -1 +1,4 @@
+Compat 8.9
```
Comment (ararslan) [inline test/REQUIRE]: Should be 0.8.9, right?

Comment (andreasnoack) [inline test/REQUIRE]: Oops.

**Comment by ararslan** [comment]
We should tag a new patch version. I can do that later today if no one beats me to it.

**Comment by tkelman** [comment]
Please do tag, the MethodError for `depwarn(::String)` broke multiple packages

File: test/deprecated.jl [hunk 2]
```diff
@@ -0,0 +1,8 @@
+using DataFrames, FileIO, RDatasets, Compat
+fn = tempname()
+open(fn, "w") do f
+    redirect_stderr(f) do
+        dataset("datasets", "iris")
+    end
+end
+@test contains(readstring(fn), "WARNING: read_rda(args...) is deprecated, use FileIO.load(args...) instead.")
```
Comment (tkelman) [inline test/deprecated.jl]: you should delete the file when this is done

**Pull request merged**